### DISCUSSION TITLE
Does parcel support optional chaining?

### DISCUSSION THREAD

**Discussion by xzilja** [parcel-bundler/parcel#4819] [General]
I'm getting following error when using optional chaining alongside latest version of typescript. Does parcel support it or is custom babelrc required?

<img width="417" alt="Screenshot 2020-06-29 at 21 29 31" src="https://user-images.githubusercontent.com/3154053/86042394-c5af3c80-ba4f-11ea-9b34-e6926b15632b.png">

**Comment by mischnic** [comment]
You might need to upgrade Babel (in your lockfile)

**Reply by xzilja** [reply]
All related babel dependencies were up to date. I was on `parcel@2.0.0-beta.1` and switched to nightly `2.0.0-nightly.318`, build passed, but is broken.

For moree context, this is how my `package.json` is set up
```json
{
  "browser": "dist/index.html",
  "source": "public/index.html",
  "engines": {
    "browsers": "Chrome 80, Safari 13.1"
  }
}
```

This is also a package inside a monorepo i.e. `packages/app`, not sure if this could be the issuee. Should I add custom `.babelrc` with appropriate plugin to it?

**Reply by xzilja** [reply]
Small note I forgot to mention. Error only happens during build command, when I `serve` the app, everything works fine. Also nullish coalescing with `??` works fine in both cases, which is even more weird.

**Reply by mischnic** [reply]
Then it's probably because of Terser, which doesn't support optional chaining yet. You need to either transpile it via Babel or not run Terser at all (which is probably worse)

**Reply by Hatko** [reply]
@IljaDaderko I experience a similar kind of issue - using OC in TS doesn't seem to work when `serve` is run. Could you tell me what you ended up doing?

**Reply by xzilja** [reply]
I don't really remember what I did with that error, but I assume it should be fixed now on all the latest versions? I no longer use parcel, so can't tell you for sure.
